<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Utilizing LIGER for the integration of spatial transcriptomic data • birsbio2020.seqfish.ligerint</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Utilizing LIGER for the integration of spatial transcriptomic data">
<meta property="og:description" content="birsbio2020.seqfish.ligerint">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=UA-93043521-1"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-93043521-1');
</script>
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">birsbio2020.seqfish.ligerint</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/birs_revised.html">Utilizing LIGER for the integration of spatial transcriptomic data</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/jsodicoff/BIRSBIO2020.seqFISH.LIGERintegration">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="birs_revised_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Utilizing LIGER for the integration of spatial transcriptomic data</h1>
                        <h4 class="author">Joshua Sodicoff</h4>
            
            <h4 class="date">7/17/2020</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/jsodicoff/birs_liger_integration/blob/master/vignettes/birs_revised.Rmd"><code>vignettes/birs_revised.Rmd</code></a></small>
      <div class="hidden name"><code>birs_revised.Rmd</code></div>

    </div>

    
    
<div id="updates" class="section level2">
<h2 class="hasAnchor">
<a href="#updates" class="anchor"></a>Updates</h2>
<p>This vignette will differ somewhat from my original presentation, namely in that it does not require access to any datasets outside of those provided for the hackathon. Likewise, instead of comparing an analysis across spatial modalities, it will demonstrate the method for generating the results required for that on the conference data. I hope this vignette helps you learn how to use Liger for this task and interpret UMAP plots and clustering metrics to determine the accuracy of your integration</p>
</div>
<div id="overview" class="section level2">
<h2 class="hasAnchor">
<a href="#overview" class="anchor"></a>Overview</h2>
<p>In this analysis, the additional structure found by combining spatial and single cell transcriptomic datasets with Liger is demonstrated. From the separate and integrative analyses, plots of identified and known clusters are generated, and metrics of integration performance are compared in context.</p>
<p>#Viewing the data</p>
<p>First, let’s preview the data to ensure that gene features comprise the rows, samples comprise the columns, and known cluster assignments are factor vectors.</p>
<div class="sourceCode" id="cb1"><html><body><pre class="r"><span class="no">tasic</span>[<span class="fl">1</span>:<span class="fl">10</span>,<span class="fl">1</span>:<span class="fl">10</span>]</pre></body></html></div>
<pre><code>##          tasic_1 tasic_2 tasic_3 tasic_4 tasic_5 tasic_6 tasic_7 tasic_8
## abca15        11      42      17      42      35      35      35      42
## abca9         22      46      22      46      39      39      39      46
## acta2         15      47      15      42      34      42      34      47
## adcy4         12      45      12      45      38      38      38      45
## aldh3b2       27      49      27      49      42      42      49      49
## amigo2        23      43     101      43      72      43      43      49
## ankle1        29      46      72      37      61      35      41      24
## ano7          11      42      16      42      34      34      34      42
## anxa9         15      48      15      42      33      42      33      42
## arhgef26      59      43      75      43      36      43      85      49
##          tasic_9 tasic_10
## abca15        48       35
## abca9         46       39
## acta2         47       42
## adcy4         45       38
## aldh3b2       49       42
## amigo2        49       43
## ankle1        49       48
## ano7          48       34
## anxa9         48       42
## arhgef26      49       54</code></pre>
<div class="sourceCode" id="cb3"><html><body><pre class="r"><span class="no">tasic_clust</span>[<span class="fl">1</span>:<span class="fl">10</span>]</pre></body></html></div>
<pre><code>##   tasic_1   tasic_2   tasic_3   tasic_4   tasic_5   tasic_6   tasic_7   tasic_8 
## Astrocyte Astrocyte Astrocyte Astrocyte Astrocyte Astrocyte Astrocyte Astrocyte 
##   tasic_9  tasic_10 
## Astrocyte Astrocyte 
## 8 Levels: Astrocyte Endothelial Cell ... Oligodendrocyte.3</code></pre>
<div class="sourceCode" id="cb5"><html><body><pre class="r"><span class="no">seqfish</span>[<span class="fl">1</span>:<span class="fl">10</span>,<span class="fl">1</span>:<span class="fl">10</span>]</pre></body></html></div>
<pre><code>##          seqfish_1 seqfish_2 seqfish_3 seqfish_4 seqfish_5 seqfish_6 seqfish_7
## abca15          68        49        50        39        31        59        44
## abca9           41        42        38        36        47        18        37
## acta2           25        23        16        21        29        36        21
## adcy4           39        54        37        18        37        26        36
## aldh3b2        101        47        41        52       101        50       101
## amigo2          93        64        93        93        93        93        63
## ankle1          41        42        76        40        46        38        63
## ano7            44        68        50        48        50        46        41
## anxa9           53        40        43        39        31        39        36
## arhgef26        38        40        42        45        46        44        37
##          seqfish_8 seqfish_9 seqfish_10
## abca15          66        50         46
## abca9           48        31         41
## acta2           32        25         11
## adcy4           32        33         40
## aldh3b2         56        40         42
## amigo2          56        93         62
## ankle1          27         9         41
## ano7            37        31         46
## anxa9           36        44         82
## arhgef26        45        28         42</code></pre>
<div class="sourceCode" id="cb7"><html><body><pre class="r"><span class="no">seq_clust</span>[<span class="fl">1</span>:<span class="fl">10</span>]</pre></body></html></div>
<pre><code>##            seqfish_1            seqfish_2            seqfish_3 
## Glutamatergic Neuron Glutamatergic Neuron Glutamatergic Neuron 
##            seqfish_4            seqfish_5            seqfish_6 
## Glutamatergic Neuron Glutamatergic Neuron Glutamatergic Neuron 
##            seqfish_7            seqfish_8            seqfish_9 
## Glutamatergic Neuron Glutamatergic Neuron Glutamatergic Neuron 
##           seqfish_10 
## Glutamatergic Neuron 
## 8 Levels: Astrocyte Endothelial Cell ... Oligodendrocyte.3</code></pre>
</div>
<div id="separate-integrations" class="section level2">
<h2 class="hasAnchor">
<a href="#separate-integrations" class="anchor"></a>Separate integrations</h2>
<p>To establish a baseline as the level of structure and clustering accuracy that can be found for each dataset with LIGER, we run individual analyses.</p>
<p>The analysis includes 1. Normalization, to standardize the counts per sample 2. Variable gene selection 3. Scaling, but not centering as not to violate the nonnegativity constraint 4. Integrative non-negative matrix factorization, or essentially just NMF in the case shown here in which one dataset is provided 5. Quantile normalization, primarily used for aligning multiple datasets. 6. Cluster identification with the Louvain algorithm 7. UMAP, for visualization</p>
<p>First, we run the analysis on the scRNA-seq data</p>
<div class="sourceCode" id="cb9"><html><body><pre class="r"><span class="no">tasic_obj</span> <span class="kw">=</span> <span class="fu">createLiger</span>(<span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span>(<span class="kw">tasic</span> <span class="kw">=</span> <span class="no">tasic</span>))
<span class="no">tasic_obj</span> <span class="kw">%&lt;&gt;%</span> <span class="kw pkg">liger</span><span class="kw ns">::</span><span class="fu"><a href="https://rdrr.io/pkg/liger/man/normalize.html">normalize</a></span>() <span class="kw">%&gt;%</span>
    <span class="kw pkg">liger</span><span class="kw ns">::</span><span class="fu"><a href="https://rdrr.io/pkg/liger/man/selectGenes.html">selectGenes</a></span>() <span class="kw">%&gt;%</span>
    <span class="kw pkg">liger</span><span class="kw ns">::</span><span class="fu"><a href="https://rdrr.io/pkg/liger/man/scaleNotCenter.html">scaleNotCenter</a></span>() <span class="kw">%&gt;%</span>
    <span class="kw pkg">liger</span><span class="kw ns">::</span><span class="fu"><a href="https://rdrr.io/pkg/liger/man/optimizeALS.html">optimizeALS</a></span>(<span class="kw">k</span> <span class="kw">=</span> <span class="fl">20</span>) <span class="kw">%&gt;%</span>
    <span class="kw pkg">liger</span><span class="kw ns">::</span><span class="fu"><a href="https://rdrr.io/pkg/liger/man/quantile_norm.html">quantile_norm</a></span>() <span class="kw">%&gt;%</span>
    <span class="kw pkg">liger</span><span class="kw ns">::</span><span class="fu"><a href="https://rdrr.io/pkg/liger/man/louvainCluster.html">louvainCluster</a></span>() <span class="kw">%&gt;%</span>
    <span class="kw pkg">liger</span><span class="kw ns">::</span><span class="fu"><a href="https://rdrr.io/pkg/liger/man/runUMAP.html">runUMAP</a></span>()</pre></body></html></div>
<pre><code>## 
  |                                                                            
  |                                                                      |   0%
  |                                                                            
  |==                                                                    |   3%
  |                                                                            
  |=====                                                                 |   7%
  |                                                                            
  |=======                                                               |  10%
  |                                                                            
  |=========                                                             |  13%
  |                                                                            
  |============                                                          |  17%
  |                                                                            
  |==============                                                        |  20%
  |                                                                            
  |================                                                      |  23%
  |                                                                            
  |===================                                                   |  27%
  |                                                                            
  |=====================                                                 |  30%
  |                                                                            
  |=======================                                               |  33%
  |                                                                            
  |==========================                                            |  37%
  |                                                                            
  |============================                                          |  40%
  |                                                                            
  |==============================                                        |  43%
  |                                                                            
  |=================================                                     |  47%
  |                                                                            
  |===================================                                   |  50%
  |                                                                            
  |=====================================                                 |  53%
  |                                                                            
  |========================================                              |  57%
  |                                                                            
  |==========================================                            |  60%
  |                                                                            
  |============================================                          |  63%
  |                                                                            
  |===============================================                       |  67%
  |                                                                            
  |=================================================                     |  70%
  |                                                                            
  |===================================================                   |  73%
  |                                                                            
  |======================================================                |  77%
  |                                                                            
  |========================================================              |  80%
  |                                                                            
  |==========================================================            |  83%
  |                                                                            
  |=============================================================         |  87%
  |                                                                            
  |===============================================================       |  90%
  |                                                                            
  |=================================================================     |  93%
  |                                                                            
  |====================================================================  |  97%
  |                                                                            
  |======================================================================| 100%
## Finished in 21.40201 secs, 30 iterations.
## Max iterations set: 30.
## Final objective delta: 0.0004846784.
## Best results with seed 1.</code></pre>
<p>Then, we run it on the seqFISH data</p>
<div class="sourceCode" id="cb11"><html><body><pre class="r"><span class="no">seq_obj</span> <span class="kw">=</span> <span class="fu">createLiger</span>(<span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span>(<span class="kw">seqfish</span> <span class="kw">=</span> <span class="no">seqfish</span>))
<span class="no">seq_obj</span> <span class="kw">%&lt;&gt;%</span> <span class="kw pkg">liger</span><span class="kw ns">::</span><span class="fu"><a href="https://rdrr.io/pkg/liger/man/normalize.html">normalize</a></span>() <span class="kw">%&gt;%</span>
    <span class="kw pkg">liger</span><span class="kw ns">::</span><span class="fu"><a href="https://rdrr.io/pkg/liger/man/selectGenes.html">selectGenes</a></span>() <span class="kw">%&gt;%</span>
    <span class="kw pkg">liger</span><span class="kw ns">::</span><span class="fu"><a href="https://rdrr.io/pkg/liger/man/scaleNotCenter.html">scaleNotCenter</a></span>() <span class="kw">%&gt;%</span>
    <span class="kw pkg">liger</span><span class="kw ns">::</span><span class="fu"><a href="https://rdrr.io/pkg/liger/man/optimizeALS.html">optimizeALS</a></span>(<span class="kw">k</span> <span class="kw">=</span> <span class="fl">20</span>) <span class="kw">%&gt;%</span>
    <span class="kw pkg">liger</span><span class="kw ns">::</span><span class="fu"><a href="https://rdrr.io/pkg/liger/man/quantile_norm.html">quantile_norm</a></span>() <span class="kw">%&gt;%</span>
    <span class="kw pkg">liger</span><span class="kw ns">::</span><span class="fu"><a href="https://rdrr.io/pkg/liger/man/louvainCluster.html">louvainCluster</a></span>() <span class="kw">%&gt;%</span>
    <span class="kw pkg">liger</span><span class="kw ns">::</span><span class="fu"><a href="https://rdrr.io/pkg/liger/man/runUMAP.html">runUMAP</a></span>()</pre></body></html></div>
<pre><code>## 
  |                                                                            
  |                                                                      |   0%
  |                                                                            
  |==                                                                    |   3%
  |                                                                            
  |=====                                                                 |   7%
  |                                                                            
  |=======                                                               |  10%
  |                                                                            
  |=========                                                             |  13%
  |                                                                            
  |============                                                          |  17%
  |                                                                            
  |==============                                                        |  20%
  |                                                                            
  |================                                                      |  23%
  |                                                                            
  |===================                                                   |  27%
  |                                                                            
  |=====================                                                 |  30%
  |                                                                            
  |=======================                                               |  33%
  |                                                                            
  |==========================                                            |  37%
  |                                                                            
  |============================                                          |  40%
  |                                                                            
  |==============================                                        |  43%
  |                                                                            
  |=================================                                     |  47%
  |                                                                            
  |===================================                                   |  50%
  |                                                                            
  |=====================================                                 |  53%
  |                                                                            
  |========================================                              |  57%
  |                                                                            
  |==========================================                            |  60%
  |                                                                            
  |============================================                          |  63%
  |                                                                            
  |===============================================                       |  67%
  |                                                                            
  |=================================================                     |  70%
  |                                                                            
  |===================================================                   |  73%
  |                                                                            
  |======================================================                |  77%
  |                                                                            
  |========================================================              |  80%
  |                                                                            
  |==========================================================            |  83%
  |                                                                            
  |=============================================================         |  87%
  |                                                                            
  |===============================================================       |  90%
  |                                                                            
  |=================================================================     |  93%
  |                                                                            
  |====================================================================  |  97%
  |                                                                            
  |======================================================================| 100%
## Finished in 21.26762 secs, 30 iterations.
## Max iterations set: 30.
## Final objective delta: 0.0003805545.
## Best results with seed 1.</code></pre>
<p>We then plot the evaluated data by generated and known cluster for the scRNA-seq</p>
<div class="sourceCode" id="cb13"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span>(<span class="no">tasic_obj</span>@<span class="kw">clusters</span>) <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span>(<span class="no">tasic</span>)
<span class="no">tasic_plots</span> <span class="kw">=</span> <span class="fu">plotByDatasetAndCluster</span>(<span class="no">tasic_obj</span>, <span class="kw">return.plots</span> <span class="kw">=</span> <span class="no">T</span>)
<span class="no">tasic_plot_known_clust</span> <span class="kw">=</span> <span class="fu">plotByDatasetAndCluster</span>(<span class="no">tasic_obj</span>, <span class="kw">clusters</span> <span class="kw">=</span> <span class="no">tasic_clust</span>, <span class="kw">return.plots</span> <span class="kw">=</span> <span class="no">T</span>)
<span class="no">tasic_plots</span><span class="kw">[[</span><span class="fl">2</span>]]</pre></body></html></div>
<p><img src="birs_revised_files/figure-html/plot_tasic-1.png" width="700"></p>
<div class="sourceCode" id="cb14"><html><body><pre class="r"><span class="no">tasic_plot_known_clust</span><span class="kw">[[</span><span class="fl">2</span>]]</pre></body></html></div>
<p><img src="birs_revised_files/figure-html/plot_tasic-2.png" width="700"> and then the spatial data</p>
<div class="sourceCode" id="cb15"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span>(<span class="no">seq_obj</span>@<span class="kw">clusters</span>) <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span>(<span class="no">seqfish</span>)
<span class="no">seq_plots</span> <span class="kw">=</span> <span class="fu">plotByDatasetAndCluster</span>(<span class="no">seq_obj</span>, <span class="kw">return.plots</span> <span class="kw">=</span> <span class="no">T</span>)
<span class="no">seq_plot_known_clust</span> <span class="kw">=</span> <span class="fu">plotByDatasetAndCluster</span>(<span class="no">seq_obj</span>, <span class="kw">clusters</span> <span class="kw">=</span> <span class="no">seq_clust</span>, <span class="kw">return.plots</span> <span class="kw">=</span> <span class="no">T</span>)
<span class="no">seq_plots</span><span class="kw">[[</span><span class="fl">2</span>]]</pre></body></html></div>
<p><img src="birs_revised_files/figure-html/plot_seq-1.png" width="700"></p>
<div class="sourceCode" id="cb16"><html><body><pre class="r"><span class="no">seq_plot_known_clust</span><span class="kw">[[</span><span class="fl">2</span>]]</pre></body></html></div>
<p><img src="birs_revised_files/figure-html/plot_seq-2.png" width="700"> We can make a few qualitative obeservations: * Neither analysis seems to have a lot of structure, resulting in the UMAP plots coming out rather “blobby”. * The known clusters in the scRNA-seq seem to map slightly better to the generated clusters than the spatial data.</p>
<p>We calculate metrics of clustering accuracy to compare. ARI is the adjusted Rand index, a metric based on the number of cell-cell pairs known to reside in the same and different clusters and adjusted to account for random chance. Purity is found by dividing the total number of correctly identified cells in a cluster over the total number of cells in that cluster.</p>
<div class="sourceCode" id="cb17"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span>(<span class="st">"tasic ARI:"</span>,<span class="fu">calcARI</span>(<span class="no">tasic_obj</span>, <span class="no">tasic_clust</span>)))</pre></body></html></div>
<pre><code>## [1] "tasic ARI: 0.124928479183329"</code></pre>
<div class="sourceCode" id="cb19"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span>(<span class="st">"tasic purity:"</span>,<span class="fu">calcPurity</span>(<span class="no">tasic_obj</span>, <span class="no">tasic_clust</span>)))</pre></body></html></div>
<pre><code>## [1] "tasic purity: 0.737086477074869"</code></pre>
<div class="sourceCode" id="cb21"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span>(<span class="st">"seqfish ARI:"</span>,<span class="fu">calcARI</span>(<span class="no">seq_obj</span>, <span class="no">seq_clust</span>)))</pre></body></html></div>
<pre><code>## [1] "seqfish ARI: 0.00922479431962868"</code></pre>
<div class="sourceCode" id="cb23"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span>(<span class="st">"seqfish purity:"</span>,<span class="fu">calcPurity</span>(<span class="no">seq_obj</span>, <span class="no">seq_clust</span>)))</pre></body></html></div>
<pre><code>## [1] "seqfish purity: 0.546023794614903"</code></pre>
<p>This confirms that the single cell data had somewhat better clustering accuracy.</p>
<p>Now, let’s try to integrate the datasets, using almost the same workflow as before but with both datasets provided for object creation.</p>
<div class="sourceCode" id="cb25"><html><body><pre class="r"><span class="no">int_obj</span> <span class="kw">=</span> <span class="fu">createLiger</span>(<span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span>(<span class="kw">tasic</span> <span class="kw">=</span> <span class="no">tasic</span>, <span class="kw">seqfish</span> <span class="kw">=</span> <span class="no">seqfish</span>))
<span class="no">int_obj</span> <span class="kw">%&lt;&gt;%</span> <span class="kw pkg">liger</span><span class="kw ns">::</span><span class="fu"><a href="https://rdrr.io/pkg/liger/man/normalize.html">normalize</a></span>() <span class="kw">%&gt;%</span>
    <span class="kw pkg">liger</span><span class="kw ns">::</span><span class="fu"><a href="https://rdrr.io/pkg/liger/man/selectGenes.html">selectGenes</a></span>() <span class="kw">%&gt;%</span>
    <span class="kw pkg">liger</span><span class="kw ns">::</span><span class="fu"><a href="https://rdrr.io/pkg/liger/man/scaleNotCenter.html">scaleNotCenter</a></span>() <span class="kw">%&gt;%</span>
    <span class="kw pkg">liger</span><span class="kw ns">::</span><span class="fu"><a href="https://rdrr.io/pkg/liger/man/optimizeALS.html">optimizeALS</a></span>(<span class="kw">k</span> <span class="kw">=</span> <span class="fl">20</span>) <span class="kw">%&gt;%</span>
    <span class="kw pkg">liger</span><span class="kw ns">::</span><span class="fu"><a href="https://rdrr.io/pkg/liger/man/quantile_norm.html">quantile_norm</a></span>() <span class="kw">%&gt;%</span>
    <span class="kw pkg">liger</span><span class="kw ns">::</span><span class="fu"><a href="https://rdrr.io/pkg/liger/man/louvainCluster.html">louvainCluster</a></span>() <span class="kw">%&gt;%</span>
    <span class="kw pkg">liger</span><span class="kw ns">::</span><span class="fu"><a href="https://rdrr.io/pkg/liger/man/runUMAP.html">runUMAP</a></span>()</pre></body></html></div>
<pre><code>## 
  |                                                                            
  |                                                                      |   0%
  |                                                                            
  |==                                                                    |   3%
  |                                                                            
  |=====                                                                 |   7%
  |                                                                            
  |=======                                                               |  10%
  |                                                                            
  |=========                                                             |  13%
  |                                                                            
  |============                                                          |  17%
  |                                                                            
  |==============                                                        |  20%
  |                                                                            
  |================                                                      |  23%
  |                                                                            
  |===================                                                   |  27%
  |                                                                            
  |=====================                                                 |  30%
  |                                                                            
  |=======================                                               |  33%
  |                                                                            
  |==========================                                            |  37%
  |                                                                            
  |============================                                          |  40%
  |                                                                            
  |==============================                                        |  43%
  |                                                                            
  |=================================                                     |  47%
  |                                                                            
  |===================================                                   |  50%
  |                                                                            
  |=====================================                                 |  53%
  |                                                                            
  |========================================                              |  57%
  |                                                                            
  |==========================================                            |  60%
  |                                                                            
  |============================================                          |  63%
  |                                                                            
  |===============================================                       |  67%
  |                                                                            
  |=================================================                     |  70%
  |                                                                            
  |===================================================                   |  73%
  |                                                                            
  |======================================================                |  77%
  |                                                                            
  |========================================================              |  80%
  |                                                                            
  |==========================================================            |  83%
  |                                                                            
  |=============================================================         |  87%
  |                                                                            
  |===============================================================       |  90%
  |                                                                            
  |=================================================================     |  93%
  |                                                                            
  |====================================================================  |  97%
  |                                                                            
  |======================================================================| 100%
## Finished in 24.68607 secs, 30 iterations.
## Max iterations set: 30.
## Final objective delta: 0.0007783069.
## Best results with seed 1.</code></pre>
<p>Again, we plot the output, including one showing dataset of origin</p>
<div class="sourceCode" id="cb27"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span>(<span class="no">int_obj</span>@<span class="kw">clusters</span>) <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span>(<span class="no">tasic</span>), <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span>(<span class="no">seqfish</span>))
<span class="no">int_clust</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">as.factor</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/character.html">as.character</a></span>(<span class="no">tasic_clust</span>), <span class="fu"><a href="https://rdrr.io/r/base/character.html">as.character</a></span>(<span class="no">seq_clust</span>))); <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span>(<span class="no">int_clust</span>) <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span>(<span class="no">tasic_clust</span>), <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span>(<span class="no">seq_clust</span>))
<span class="no">int_plots</span> <span class="kw">=</span> <span class="fu">plotByDatasetAndCluster</span>(<span class="no">int_obj</span>, <span class="kw">return.plots</span> <span class="kw">=</span> <span class="no">T</span>)
<span class="no">int_plot_known_clust</span> <span class="kw">=</span> <span class="fu">plotByDatasetAndCluster</span>(<span class="no">int_obj</span>, <span class="kw">clusters</span> <span class="kw">=</span> <span class="no">int_clust</span>, <span class="kw">return.plots</span> <span class="kw">=</span> <span class="no">T</span>)
<span class="no">int_plots</span><span class="kw">[[</span><span class="fl">1</span>]]</pre></body></html></div>
<p><img src="birs_revised_files/figure-html/plot_int-1.png" width="700"></p>
<div class="sourceCode" id="cb28"><html><body><pre class="r"><span class="no">int_plots</span><span class="kw">[[</span><span class="fl">2</span>]]</pre></body></html></div>
<p><img src="birs_revised_files/figure-html/plot_int-2.png" width="700"></p>
<div class="sourceCode" id="cb29"><html><body><pre class="r"><span class="no">int_plot_known_clust</span><span class="kw">[[</span><span class="fl">2</span>]]</pre></body></html></div>
<p><img src="birs_revised_files/figure-html/plot_int-3.png" width="700"></p>
<p>We calculate clustering accuracy, as well as agreement and alignment</p>
<div class="sourceCode" id="cb30"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span>(<span class="st">"overall ARI:"</span>,<span class="fu">calcARI</span>(<span class="no">int_obj</span>, <span class="no">int_clust</span>)))</pre></body></html></div>
<pre><code>## [1] "overall ARI: 0.03990832682262"</code></pre>
<div class="sourceCode" id="cb32"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span>(<span class="st">"overall purity:"</span>,<span class="fu">calcPurity</span>(<span class="no">int_obj</span>, <span class="no">int_clust</span>)))</pre></body></html></div>
<pre><code>## [1] "overall purity: 0.621084337349398"</code></pre>
<div class="sourceCode" id="cb34"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span>(<span class="st">"agreement:"</span>,<span class="fu">calcAgreement</span>(<span class="no">int_obj</span>)))</pre></body></html></div>
<pre><code>## [1] "Reducing dimensionality using NMF"
## 
## Converged in 0.03119183 seconds, 167 iterations. Objective: 5408.016 
## 
## Converged in 0.01372457 seconds, 188 iterations. Objective: 8577.805 
## [1] "agreement: 0.089592886480605"</code></pre>
<div class="sourceCode" id="cb36"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span>(<span class="st">"alignment:"</span>,<span class="fu">calcAlignment</span>(<span class="no">int_obj</span>)))</pre></body></html></div>
<pre><code>## [1] "alignment: 0.804064439004953"</code></pre>
<p>This seems to demonstrate that the clustering accuracy of the integration (a surrogate for how well the method divided the cell types) falls somehwere between solely using the spatial data and scRNA-seq. Additionally, the agreement metric is relatively low (on its scale of 0 to 1). This implies that the integration was not very informative and relied heavily on distortion of the data for alignment.</p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Joshua Sodicoff.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.5.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
